### PR TITLE
Platooning model

### PR DESCRIPTION
**Pull request by schillic**
There is a statement in the ARCH-COMP report which IMO is wrong.

> There seems to be no reason why the upper bound tc should be greater than tb. Loss of communication after a time longer than tb is included by reestablishing communication in zero time. Tools could possibly exploit this by minimizing the model before the analysis.

I claim that for tb = 10 and tc = 10, one cannot get to the rhs location at time 15.

### CODE REVIEW

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+- `deterministic_switching` -- (optional, default: `true`) flag to use
+                               deterministic transitions
+- `time_horizon`            -- (optional, default: `20.`) time horizon
+- `allowed_distance`        -- (optional, default: `50.`) specification of
+                               allowed distance between the vehicles
+"""
+function platooning(deterministic_switching::Bool=true,
+                    time_horizon::Float64=20.,
+                    allowed_distance::Float64=50.)
+    n = 9 + 1
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    n = 9 + 1  # for each vehicle, (ei, d(et)/dt, ai) for (spacing error, relative velocity, speed) and the last dimension is time 
```

Comment (schillic) [reply models/ARCH/AFF/Platooning/Platooning.jl]: Okay, will commit together with keyword args!

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    # transition graph
+    automaton = LightAutomaton(2)
+    add_transition!(automaton, 1, 2, 12)
+    add_transition!(automaton, 2, 1, 21)
+
+    # common inputs
+    B = sparse([2], [1], [1.], n, 1)
+    BB = Matrix(1.0I, size(B, 1), size(B, 1))
+    U = Interval(-9., 1.)
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    U = Interval(-9., 1.) # acceleration of the leader truck
```

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+"""
+function platooning(;
+                    deterministic_switching::Bool=true,
+                    time_horizon::Float64=20.,
+                    allowed_distance::Float64=50.)
+    # three variables for each vehicle, (ei, d(et)/dt, ai) for
+    # (spacing error, relative velocity, speed), and the last dimension is time
+    n = 9 + 1
+
+    c1 = c2 = 5.
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    c1 = c2 = 5.   # clock constraints
```

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    tb = 10.
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    tb = 10.  # lower bound for loss of communication
```

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    tc = tr = 20.
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    tc = tr = 20.  # upper bound for loss of communication (tc) and reset time (tr)
```

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    add_transition!(automaton, 1, 2, 12)
+    add_transition!(automaton, 2, 1, 21)
+
+    # common inputs
+    B = sparse([2], [1], [1.], n, 1)
+    BB = Matrix(1.0I, size(B, 1), size(B, 1))
+    U = Interval(-9., 1.)  # acceleration of the lead vehicle
+
+    # common invariant
+    X = HPolyhedron([HalfSpace(sparsevec([n], [1.], n), 20.)])
```
Comment (schillic) [inline models/ARCH/AFF/Platooning/Platooning.jl]: Replace this invariant and use a `Universe` instead.

File: models/ARCH/AFF/Platooning/Platooning.jl [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    ℋ = HybridSystem(automaton, modes, resetmaps, switchings)
+
+    # initial condition in mode 1
+    X0 = Singleton(zeros(n))
+    initial_condition = [(1, X0)]
+
+    system = InitialValueProblem(ℋ, initial_condition)
+
+    # safety property
+    d1 = zeros(n); d1[1] = -1.
```
Comment (mforets) [inline models/ARCH/AFF/Platooning/Platooning.jl]:
```suggestion
    d1 = zeros(n); d1[1] = -1.  # x1 => -dmin
```

**Pull request merged**